flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/reader/IterationReader.java

```diff
@@ -21,34 +21,21 @@
 import org.apache.flink.runtime.event.TaskEvent;
 import org.apache.flink.runtime.util.event.EventListener;
 
-import java.io.IOException;
-
 /**
- * The basic API for every reader.
+ * Reader for iteration.
  */
-public interface ReaderBase {
-
-	/**
-	 * Returns whether the reader has consumed the input.
-	 */
-	boolean isFinished();
-
-	// ------------------------------------------------------------------------
-	// Task events
-	// ------------------------------------------------------------------------
-
-	void sendTaskEvent(TaskEvent event) throws IOException;
-
-	void registerTaskEventListener(EventListener<TaskEvent> listener, Class<? extends TaskEvent> eventType);
-
-	// ------------------------------------------------------------------------
-	// Iterations
-	// ------------------------------------------------------------------------
+public interface IterationReader {
 
 	void setIterativeReader();
 
 	void startNextSuperstep();
 
 	boolean hasReachedEndOfSuperstep();
 
+	/**
+	 * Returns whether the reader has consumed the input.
+	 */
+	boolean isFinished();
```

Comment:
I think `isFinished` method is not strong correlation of iteration, and it should be a very common method suitable for all readers.

Reply:
well, true from literal, however, isFinished() is only used in iteration scenario. maybe we can also change the name of it.

Reply:
Yes, I just confirmed it is actually only used in iteration, so all the current methods in `ReaderBase` are for iteration. If we change it to `IterationReaderBase`, the corresponding `AbstractReader` should also be changed to `AbstractIterationReader`, because the current `AbstractReader` is also only for iteration.

But I think it still seems a little strange to let `IterationReaderBase` in the upper layer, because the current `Reader` interface which is used for common read extends the `IterationReaderBase`. This relationship seems not make sense.

If we have two independent reader base interfaces, and one is for iteration, the other is for common case. Then the specific record reader can implement both of interfaces for iteration scenario or only implement common reader interface for non-iteration scenario.

So it may need reorganize the current relationships if we want to change this part. Easily change the name to `IterationReaderBase` may bring literal confusing.